release/src/router/rc/wanduck.c

```diff
@@ -2371,7 +2372,7 @@ void handle_dns_req(int sfd, unsigned char *request, int maxlen, struct sockaddr
 		strncat(queries.name, (char *)ptr, len);
 		ptr += len;
 	}
-	if (queries.type == 0 || queries.ip_class == 0 || strlen(queries.name) > 1025)
+	if (queries.type == 0 || queries.ip_class == 0 || strlen(queries.name) > PATHLEN)
```

Comment:
1025 is max hostname length from standard, although name is PATHLEN==256 sized, better to keep it in case of PATHLEN will be increased.

Reply:
I don't feel it's wanduck's job to validate if DNS queries conform to the RFC or not, but that's just a personal opinion.

Reply:
since it implements dns stub, it must follow related dns protocol rules incl. validation